Tribler/Core/Modules/restapi/channels/base_channels_endpoint.py

```diff
@@ -0,0 +1,76 @@
+import json
+import logging
+import time
+from twisted.web import http, resource
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST
+from Tribler.community.allchannel.community import AllChannelCommunity
+
+
+UNKNOWN_CHANNEL_RESPONSE_MSG = "the channel with the provided cid is not known"
+UNAUTHORIZED_RESPONSE_MSG = "you are not authorized to perform this request"
+
+
+class BaseChannelsEndpoint(resource.Resource):
+    """
+    This class contains some utility methods to work with raw channels from the database.
+    All endpoints that are using the database, should derive from this class.
+    """
+
+    def __init__(self, session):
+        resource.Resource.__init__(self)
+        self.session = session
+        self.channel_db_handler = self.session.open_dbhandler(NTFY_CHANNELCAST)
+        self._logger = logging.getLogger(self.__class__.__name__)
+
+    @staticmethod
+    def return_404(request, message=UNKNOWN_CHANNEL_RESPONSE_MSG):
+        """
+        Returns a 404 response code if your channel has not been created.
+        """
+        request.setResponseCode(http.NOT_FOUND)
+        return json.dumps({"error": message})
+
+    def return_500(self, request, exception):
```

Comment:
This is improved here https://github.com/Tribler/tribler/pull/2242/commits/a1ebfecfa1d8448fd2a7cb7f845eef2c6f2a3ada